test/reshape.jl

```diff
@@ -737,6 +737,83 @@ end
     df = DataFrame(x=[:one, :two, :one], y=[1, 2, 3])
     @test_throws ArgumentError unstack(df, :x, :y)
     @test unstack(df, :x, :y, allowduplicates=true) == DataFrame(one=3, two=2)
+    @test unstack(df, :x, :y, valuestransform=identity) ==
+          DataFrame(one=[[1, 3]], two=[[2]])
+    @test unstack(df, :x, :y, valuestransform=last) ==
+          DataFrame(one=3, two=2)
+    @test unstack(df, :x, :y, valuestransform=first) ==
+          DataFrame(one=1, two=2)
+    @test unstack(df, :x, :y, valuestransform=length) ==
+          DataFrame(one=2, two=1)
+end
+
+@testset "allowduplicates as function" begin
+    df = DataFrame(rowid=[1, 1, 1, 1, 2, 2], colid=[1, 1, 2, 2, 3, 3], values=1:6)
+    @test_throws ArgumentError unstack(df, :rowid, :colid, :values)
+    @test unstack(df, :rowid, :colid, :values, allowduplicates=true) ≅
+          DataFrame("rowid" => 1:2, "1" => [2, missing],
+                    "2" => [4, missing], "3" => [missing, 6])
+    @test unstack(df, :rowid, :colid, :values, allowduplicates=true, fill=0) ==
+          DataFrame("rowid" => 1:2, "1" => [2, 0],
+                    "2" => [4, 0], "3" => [0, 6])
+    @test unstack(df, :rowid, :colid, :values, valuestransform=identity) ≅
+          DataFrame("rowid" => 1:2, "1" => [1:2, missing],
+                    "2" => [3:4, missing], "3" => [missing, 5:6])
+    @test unstack(df, :rowid, :colid, :values,
+                  valuestransform=identity, fill=Int[]) ==
+          DataFrame("rowid" => 1:2, "1" => [1:2, []],
+                    "2" => [3:4, []], "3" => [[], 5:6])
+    @test unstack(df, :rowid, :colid, :values, valuestransform=sum) ≅
+          DataFrame("rowid" => 1:2, "1" => [3, missing],
+                    "2" => [7, missing], "3" => [missing, 11])
+    @test unstack(df, :rowid, :colid, :values, valuestransform=sum, fill=0) ==
+          DataFrame("rowid" => 1:2, "1" => [3, 0],
+                    "2" => [7, 0], "3" => [0, 11])
+    @test unstack(df, :rowid, :colid, :values, valuestransform=length) ≅
+          DataFrame("rowid" => 1:2, "1" => [2, missing],
+                    "2" => [2, missing], "3" => [missing, 2])
+    @test unstack(df, :rowid, :colid, :values, valuestransform=length, fill=0) ==
+          DataFrame("rowid" => 1:2, "1" => [2, 0],
+                    "2" => [2, 0], "3" => [0, 2])
+    @test unstack(df, :rowid, :colid, :values,
+                  valuestransform=x -> isempty(x) ? missing : length(x)) ≅
+          DataFrame("rowid" => 1:2, "1" => [2, missing],
+                    "2" => [2, missing], "3" => [missing, 2])
+    @test unstack(df, :rowid, :colid, :values,
+                  valuestransform=x -> isempty(x) ? missing : x) ≅
+          DataFrame("rowid" => 1:2, "1" => [1:2, missing],
+                    "2" => [3:4, missing], "3" => [missing, 5:6])
```

Comment:
Do we have a test somewhere that checks that `fill` is used without being passed to `valuestransform`?

Reply:
I have added more tests